engine/Assets/Scripts/Modes/MatchMode/MatchResultsTracker.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+
+public class MatchResultsTracker {
+    public readonly Dictionary<Type, ITrackedData> MatchResultEntries = new Dictionary<Type, ITrackedData>();
```

Comment:
I would consider just using a List instead of a dictionary because you can get the type object by doing `GetType()`. Depends on how you use it I suppose

Reply:
Let me know you're thoughts and then I'll merge.